kashit.py

```diff
@@ -0,0 +1,31 @@
+import functools
+
+arguments_and_results = {}
+
+
+def preprocessed_results(func):
+    @functools.wraps(func)
+    def wrapper_preprocessed_results(*args, **kwargs):
+        kwargs_tuples = tuple(kwargs.items())  # convert dict to tuple to hash it

```

Comment:
What will happen if we change the order of the `kwargs`?
Is the dict always ordered?

Reply:
Understood the problem, fixed :)